src/shared/json.c

```diff
@@ -1066,6 +1066,42 @@ double json_variant_real(JsonVariant *v) {
         return 0.0;
 }
 
+int json_variant_bytes(JsonVariant *v, uint8_t **p, size_t *n) {
+        assert_return(p, -EINVAL);
+        assert_return(n, -EINVAL);
+
+        if (!v) {
+                *n = 0;
+                return 0;
+        }
+        if (!json_variant_is_array(v))
+                return -EINVAL;
+
+        if (json_variant_elements(v) == 0) {
```

Comment:
Please cache the result of `json_variant_elements()`, as it is called again below.